### ISSUE TITLE
lint: yodaStyleExpr considers if '0' <= c && c <= '9' as non-idiomatic

### ISSUE THREAD

**Issue by nightlyone** [go-critic/go-critic#905]
The Go team highly mandates to express a value range check as 
```go
 if '0' <= c && c <= '9' { 
	// character range ok
}
```
instead of 
```go
 if c >= '0'  && c <= '9' { 
	// character range ok
}
```
so the yodaStyleExpr should respect this and cease to warn about that.

NOTE: This like an regression that might have been introduced by PR #856 while trying to fix issue #355 before.

**Comment by cristaloleg** [comment]
@quasilyte let's discuss this once more ;D

**Comment by nightlyone** [comment]
@cristaloleg could you link to the previous discussion here, unless it happened offline?

**Comment by cristaloleg** [comment]
@nightlyone it happened offline-ish long time ago, but I remember we had a discussion which pattern is proper. Hopefully we can rebuild previous discussion.

Probably this might be solved via checker's config.

**Comment by quasilyte** [comment]
I personally don't like the suggested option. :)

https://github.com/quasilyte/go-consistent/ checks that you use one of these two consistently.

**Comment by nightlyone** [comment]
Has there be any outcome of the discussion?

To be clear: It does not make sense to me to nag the user about a style that is mandated by the Go team as good Go code (and requested from the Go team when missing).

It might help to limit the scope of the yodastyle linter by default to equality. That style is considered bad style by way more people than the Go team styled range checks.

So maybe making anything but equality and non-equality opt in for this linter. Or at least opt-opt.

go-consistent is also a great tool, but not actually helpful when the existing code base is from people who used Go for the first time and also learned to program in a consistent way within that code base.

It just takes the bad examples from the past and suggests to make the code worse again in that scenario. So it does not solve the problem.

**Comment by quasilyte** [comment]
@cristaloleg which solution do you like more:
1. Recognize range checks and exclude them (small % of false positive if we don't recognize something).
2. Restrict yoda to `==` and `!=` by default again (more false negatives by default).

**Comment by quasilyte** [comment]
If someone will send a patch before I have my weekend to work on this it's fine too. :)

**Comment by quasilyte** [comment]
> The Go team highly mandates to express a value range check as

@nightlyone could you provide a link to that suggestion? It would be useful for the future reference.

**Comment by dgryski** [comment]
I've tried to link to all the official and semi-official ones or know about here: https://medium.com/@dgryski/idiomatic-go-resources-966535376dba

A good starting place before search golang-nuts. The style allowed in the stdlib and suggested in CLs probably also count but may be harder to track down.
